Examples/src/main/kotlin/com/mapbox/vision/examples/UsbVideoSourceActivityKt.kt

```diff
@@ -0,0 +1,229 @@
+            VisionManager.setModelPerformanceConfig(
+                ModelPerformanceConfig.Merged(
+                    ModelPerformance.On(ModelPerformanceMode.FIXED, ModelPerformanceRate.HIGH)
+                )
+            )
+            vision_view.setVisionManager(VisionManager)
+            VisionManager.visionEventsListener = visionEventsListener
+            VisionManager.start()
+
+            visionManagerWasInit = true
+        }
+    }
+
+    private fun stopVisionManager() {
+        if (visionManagerWasInit) {
+            VisionManager.stop()
+            VisionManager.destroy()
+
+            visionManagerWasInit = false
+        }
+    }
+
+    private val onDeviceConnectListener: USBMonitor.OnDeviceConnectListener =
+        object : USBMonitor.OnDeviceConnectListener {
+            override fun onAttach(device: UsbDevice?) {
+                toast("Usb camera attached")
+                usbMonitor?.requestPermission(device!!)
+            }
+
+            override fun onConnect(
+                device: UsbDevice?,
+                ctrlBlock: USBMonitor.UsbControlBlock?,
+                createNew: Boolean
+            ) {
+                toast("Usb camera connected")
+                backgroundHandler.post {
+                    synchronized(this@UsbVideoSourceActivityKt) {
+                        uvcCamera?.destroy()
+                        uvcCamera = initializeUvcCamera(ctrlBlock!!)
+                    }
+                }
+            }
+
+            override fun onDetach(device: UsbDevice?) {
+                toast("Usb camera detached")
+            }
+
+            override fun onCancel(p0: UsbDevice?) {
+                toast("Usb camera cancelled")
+            }
+
+            override fun onDisconnect(device: UsbDevice?, ctrlBlock: USBMonitor.UsbControlBlock?) {
+                toast("Usb camera disconnected")
+                backgroundHandler.post {
+                    uvcCamera?.close()
+                }
+            }
+        }
+
+    private fun initializeUvcCamera(ctrlBlock: USBMonitor.UsbControlBlock) = UVCCamera().also { camera ->
+        camera.open(ctrlBlock)
+        camera.setPreviewSize(
+            IMAGE_SIZE.imageWidth,
+            IMAGE_SIZE.imageHeight,
+            UVCCamera.FRAME_FORMAT_MJPEG
+        )
+
+        val surfaceTexture = SurfaceTexture(createExternalGlTexture())
+        surfaceTexture.setDefaultBufferSize(
+            IMAGE_SIZE.imageWidth,
+            IMAGE_SIZE.imageHeight
+        )
+        // Start preview to external GL texture
+        // NOTE : this is necessary for callback passed to [UVCCamera.setFrameCallback]
+        // to be triggered afterwards
+        camera.setPreviewTexture(surfaceTexture)
+        camera.startPreview()
+
+        // Set callback that will feed frames from the USB camera to Vision SDK
+        camera.setFrameCallback(
+            { frame ->
+                videoSourceListener?.onNewFrame(
```

Comment:
VideoSourceListener is a bit confusing from pure Android development side. At least I expect that I'll be listening for something :)
Maybe VisionVideoSourceProvider is better, or something around this idea

Reply:
Why do you think it's confusing? It's plain listener that listens to some events (frames, in this case). 

IMO provider is smth active, some object that you ask to provide other objects

Reply:
Provider seems also confusing.
At first glance I expected that listener methods will be invoked by some code to give me data, but this listener is used to notify VisionCore about new frames (and camera params). It is ok too, but a bit unusual from my point when I try to consume new SDK and use "listeners"

Reply:
Yeah, here Vision plays as a listener itself, since user code is implementing some custom video source, reads frames from anywhere, then this code needs to feed this frames back to Vision. To do so we've created an interface that listens for frames/cameraParams. 

We hadn't found any other clear options to implement custom modules from the SDK perspective when we've implemented custom video source, do you maybe have other thoughts here?

Reply:
Good thing is - implementing your own video source is not a common requirement, most users are good with the default Camera2VideoSourceImpl we use by default

Reply:
This "event stream -> subscription" pattern looks good to me, can't propose anything better for now. Probably we can play with naming here, ExternalVideoSourceListener, VideoSourceSubscriber, ExternalVideoSourcePipe

Reply:
IMHO current naming is fine. Moreover renaming will lead to quite massive refactoring, using deprecation mechanism for old `VideoSourceListener` for at least one release. Seems like an overkill.

Reply:
@kiryldz we definitely won't change naming right now - but if there are some great ideas for better naming it's possible to rename them in 1+ releases

Reply:
@yuryybk I can certainly change the name of the variable here - wdyt about
`UsbVideoSourceListener` ? Liked your idea about adding prefix `External` but seems `Usb` is even better here

Reply:
👍 , Usb* is clear too